base/bin/build.sh

```diff
@@ -1,27 +1,188 @@
+#!/bin/bash
 
-# Change the current working directory to the `base` directory instead of
-# the `base/bin` directory
-pushd `dirname $0`/.. > /dev/null
+# fail-fast
+set -e
 
-RUNTIME_NAME="nodejs"
-DOCKER_NAMESPACE=${1}
-CANDIDATE_NAME=${2}
+function fail() {
+  echo "ERROR: ${1}"
+  exit 1
+}
+
+function print() {
+  if [ "${VERBOSE}" = "true" ]; then
+    echo "${1}"
+  fi
+}
+
+_NODE_VERSION_OPTION="--node-version"
+_UPLOAD_TO_STAGING_OPTION="--upload-to-staging"
+_CANDIDATE_NAME_OPTION="--candidate-name"
+_DOCKER_NAMESPACE_OPTION="--docker-namespace"
+_VERBOSE_OPTION="--verbose"
+_HELP_OPTION="--help"
+
+_DEFAULT="default"
```

Comment:
It sounds like not specifying an input or specifying `default` result in the same behavior. Is there a reason to have to support both?